### PR TITLE
Add Support for Passing Instance Profile to EC2 Instance Creation

### PR DESCRIPTION
**Pull request by d3sch41n**
Tests pass: 

```
➜  ec2 git:(sfm-instance-profile) ✗ go test
Successfully created instance: i-088f647ed0362fa0d
2022/06/09 10:54:58 Running instance IDs:
i-0ce6f9735343a44f6
i-04d9c931f1982cd4b
i-04105b2d07fec860f

Successfully grabbed public IP: 44.226.198.78
Successfully grabbed instance state: running
PASS
ok  	github.com/l50/awsutils/pkg/ec2	200.695s
➜  ec2 git:(sfm-instance-profile) ✗
```

### CODE REVIEW

File: pkg/ec2/ec2.go [hunk 1]
```diff
@@ -69,6 +70,9 @@ func CreateInstance(client *ec2.EC2, ec2Params Params) (*ec2.Reservation, error)
 				},
 			},
 		},
```
Comment (l50) [inline pkg/ec2/ec2.go:72]: Let's capitalize all instances of Iam. For example: `IamInstanceProfile` -> `IAMInstanceProfile`

File: pkg/ec2/test_env [hunk 2]
```diff
@@ -1,7 +1,8 @@
-export AMI=TODO
+export AMI=ami-0ca285d4c2cda3300
```
Comment (l50) [inline pkg/ec2/test_env]: This file is meant to serve as an example, so let's drop any changes to this file from the PR.

**Review by l50 — CHANGES_REQUESTED** [review]
Thanks for the great contribution! Please see my comments for changes that are needed to land.

File: pkg/ec2/ec2.go (hunk 1 above)
Comment (d3sch41n) [reply pkg/ec2/ec2.go:72]: These are the AWS go sdk parameters/structures - I don't think we can change them

File: pkg/ec2/test_env (hunk 2 above)
Comment (d3sch41n) [reply pkg/ec2/test_env]: Done - I added a README and suffixed the example file with `_template`

**Review by l50 — APPROVED** [review]
LGTM, thanks for the contribution! Assuming everything passes the github actions run, we should be in good shape!

File: pkg/ec2/ec2.go (hunk 1 above)
Comment (l50) [reply pkg/ec2/ec2.go:72]: Whoopsie, good catch! Looks like they didn't adhere to best practices, but it's not our problem :)

**Pull request merged**